### PR TITLE
Azure network plugin part four

### PR DESCRIPTION
**Pull request by fgan1**
## Description
Implementing Azure Network Plugin
## Proposed changes
Create Virtual Network with tags and Implementing PendingInstanceManagement(Temporary Soluction)
## Additional information
This branch depends on a branch in the Fogbow Common (azure-network-plugin)
## PR Slices
[x] One: Request instance context
[x] Two: Get instance context
[x] Three: Delete instance context
[x] Four: Add tags in the Request and PendingInstanceManagement
## Reminding
PRs order:
develop < azure-network-plugin < PR2 < *PR3 < PR4*

### CODE REVIEW

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/AzureNetworkPluginTest.java [hunk 1]
```diff
@@ -96,29 +106,40 @@ public void testHasFailedSuccessfullyWhenIsFailed() {
     public void testRequestInstanceSuccessfully() throws FogbowException {
         // set up
         String cidr = "10.10.10.10/24";
-        String name = "name";
+        String resourceName = "resourceName";
```
[the remark this answers is not in the document]
Comment (jadsonluan) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/AzureNetworkPluginTest.java]: Done.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureVirtualNetworkOperationSDKTest.java [hunk 2]
```diff
@@ -67,11 +65,15 @@ public void setUp() {
     }
 
     // test case: When calling the buildVirtualNetworkCreationObservable method and the observables execute
-    // without any error, it must verify if It returns the right logs.
+    // without any error, it must verify if It returns the right logs and define instance as ready.
     @Test
     public void testBuildVirtualNetworkCreationObservableSuccessfully() {
         // set up
-        AzureCreateVirtualNetworkRef azureCreateVirtualNetworkRef = Mockito.mock(AzureCreateVirtualNetworkRef.class);
+        String resourceName = "resourceName";
```
[the remark this answers is not in the document]
Comment (jadsonluan) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureVirtualNetworkOperationSDKTest.java]: Done.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureVirtualNetworkOperationSDK.java [hunk 3]
```diff
@@ -104,15 +113,15 @@ private void subscribeVirtualNetworkCreation(Observable<Indexable> virtualNetwor
                 .subscribe();
     }
 
-    public AzureGetVirtualNetworkRef doGetInstance(String resourceName, AzureUser azureUser)
+    public AzureGetVirtualNetworkRef doGetInstance(String resourceName, String orderName, AzureUser azureUser)
```
Comment (wassoares) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureVirtualNetworkOperationSDK.java:116]: It is not necessary to pass the orderName by parameter. The resource name is contained in the resource tag.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureVirtualNetworkOperationSDK.java [hunk 4]
```diff
@@ -104,15 +113,15 @@ private void subscribeVirtualNetworkCreation(Observable<Indexable> virtualNetwor
...
             throws FogbowException {
 
         Network network = getNetwork(resourceName, azureUser);
 
         VirtualNetworkInner virtualNetworkInner = network.inner();
         String provisioningState = virtualNetworkInner.provisioningState();
         String id = virtualNetworkInner.id();
-        String name = network.name();
+        String name = orderName;
```
Comment (wassoares) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureVirtualNetworkOperationSDK.java:124]: Obtain the resource name from the resource tag, as follows: network.tags().get(AzureConstants.TAG_NAME);

Comment (jadsonluan) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureVirtualNetworkOperationSDK.java]: Done.

**Pull request merged**